docs/advanced_usage/advanced_tests.rst

```diff
@@ -15,44 +20,57 @@ an additional reference can be added into the test_run decorator:
+collide with default command line option names.
 
-.. note:: Cli tags must be given in pairs. If one key has multiple values separate them with a comma.
+The only limitation is to always specify you tags as options in the CLI, i.e. as pairs of tag name
+and tag value. Multiple tag values for a single tag name simply need to be comma-separated (without whitespace).
 
-.. note:: The cli will only allow you to use the character ``-`` instead of ``_`` for tags. If you call for example 
+.. note:: The cli will only allow you to use the character ``-`` instead of ``_`` for tags. If you call for example
     ``--branch-level`` in the cli, you can use following corresponding tags in your test case: ``branch-level``,
     ``branch_level`` or ``branchlevel``.
 
+Find below an example of such a CLI invocation:
+
 .. code:: bash
 
     pykiso -c configuration_file --variant var1,var2 --branch-level daily,nightly
 
-.. table:: Exectuion table for test case tags and cli tag arguments
+
+.. table:: Execution table for test case tags and cli tag arguments
    :widths: auto
 
-   =======================================================  ===============================  ========
-   test case tags                                           cli tags                         executed
-   =======================================================  ===============================  ========
-   "branch_level": ["daily","nightly"]                      branch-level nightly             🗸
-   "branch-level": ["daily","nightly"]                      branch-level nightly             🗸
-   "branch_level": ["daily","nightly"]                      branch-level nightly,daily       🗸
-   "branch_level": ["daily","nightly"]                      branch-level master              ✗
-   "branch_level": ["daily","nightly"],"variant":["var1"]   branch-level daily variant var9  🗸
-   "branch_level": ["daily","nightly"],"variant":["var1"]   branch-level nightly             🗸
-   "branch_level": ["daily","nightly"],"variant":["var1"]   variant var1                     🗸
-   "branch_level": ["daily","nightly"],"variant":["var1"]   variant var2                     ✗
-   "branch_level": ["daily","nightly"],"variant":["var1"]   branch-level daily variant var1  ✗
-   "variant":["var1"]                                       branch-level daily               ✗
-   =======================================================  ===============================  ========
+   =============================================================  ========================================  ========
+   Test case Tags                                                 CLI Tags                                  Executed
+   =============================================================  ========================================  ========
+   ``"variant": ["var1"]``                                        ``--branch-level daily``                  🗸
```

Comment:
why would this one get executed ? 
you selected branch-level daily only

Reply:
I would prefer the old logic as it is easier to explain to a user:
"It executes the test cases where all tags mentioned on cli match."

Reply:
IMHO this is the most user-unfriendly logic that we could apply

Matching all CLI-provided tags will mean that as soon as you'll decide to add a new tag, you'll have to adapt each one of your tests. 
When you end up having huge test suites, this can take a lot of time for a feature that (as I understood it), is meant to select specific tests based on what is provided to the CLI, not filter them based on the CLI tags (that's the main difference between the point of views I believe)

Also, for test filtering, we already have the pattern option that really allows to run only the wanted tests

Reply:
So those tags on test cases are restrictions. If you add more tags on your test cases, less test cases will be executed.

Reply:
From now on, if a user adds a new tag in the CLI, he will have to specify it in all of his test cases that define tags